src/code/z_player_lib.c

```diff
@@ -1798,49 +1987,399 @@ void Player_DrawZoraShield(PlayState* play, Player* player) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80125D4C.s")
+                    if (handIndex != 0) {
+                        handIndex = (handIndex >> 8) - 1;
+                        rightHandDLists = &D_801C0964[handIndex][D_801F59E0];
+                    }
+                }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_801262C8.s")
+                *dList = rightHandDLists[sPlayerLod];
+                if (player->skelAnime.animation == &gPlayerAnim_pg_punchB) {
+                    func_80125CE0(player, D_801C0784, pos, rot);
+                }
+            }
+        } else if (limbIndex == PLAYER_LIMB_SHEATH) {
+            Gfx** sheathDLists = player->sheathDLists;
+
+            if (player->transformation == PLAYER_FORM_HUMAN) {
+                EquipValueSword swordEquipValue = GET_CUR_EQUIP_VALUE(EQUIP_TYPE_SWORD);
+
+                if (swordEquipValue != EQUIP_VALUE_SWORD_NONE) {
+                    if ((player->sheathType == PLAYER_MODELTYPE_SHEATH_14) ||
+                        (player->sheathType == PLAYER_MODELTYPE_SHEATH_12)) {
+                        sheathDLists = &gPlayerSheathedSwords[2 * ((swordEquipValue - 1) ^ 0)];
+                    } else {
+                        sheathDLists = &gPlayerSwordSheaths[2 * ((swordEquipValue - 1) ^ 0)];
+                    }
+                }
+            }
 
-// unused
-s32 func_801263FC(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+            *dList = sheathDLists[sPlayerLod];
+        } else if (limbIndex == PLAYER_LIMB_WAIST) {
+            *dList = player->waistDLists[sPlayerLod];
+        } else if (limbIndex == PLAYER_LIMB_HAT) {
+            if (player->transformation == PLAYER_FORM_ZORA) {
+                Matrix_Scale((player->unk_B10[0] * 1) + 1.0f, 1.0f, 1.0f, MTXMODE_APPLY);
+            }
+        }
+    }
+
+    return false;
+}
+
+s32 Player_OverrideLimbDrawGameplayFirstPerson(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                                               Actor* actor) {
+    Player* player = (Player*)actor;
+
+    if (!Player_OverrideLimbDrawGameplayCommon(play, limbIndex, dList, pos, rot, actor)) {
+        if (player->unk_AA5 != 3) {
+            *dList = NULL;
+        } else if (limbIndex == PLAYER_LIMB_LEFT_FOREARM) {
+            *dList = D_801C0294[player->transformation];
+        } else if (limbIndex == PLAYER_LIMB_LEFT_HAND) {
+            *dList = D_801C02A8[player->transformation];
+        } else if (limbIndex == PLAYER_LIMB_RIGHT_SHOULDER) {
+            *dList = D_801C02BC[player->transformation];
+        } else if (limbIndex == PLAYER_LIMB_RIGHT_HAND) {
+            if (Player_IsHoldingHookshot(player)) {
+                *dList = D_801C02E4[player->transformation];
+            } else {
+                *dList = D_801C02D0[player->transformation];
```

Comment:
Ah, that's what all those arrays are for. Could we name them based on their use in first-person? (Didn't see any other uses so this does seem to be their only use.)

Reply:
`sPlayerFirstPersonHookshotRightHandDLs` and `sPlayerFirstPersonRightHandDLs`?

Reply:
Yes, and the others in here. But shouldn't all the right hands start with `RightHand` and then say what they are (empty, open, fist, with hookshot, etc.)?

Reply:
Actually, I'm not sure about naming the other ones. Do we know if they are used for first person or not?
Naming those will depend in what `player->unk_AA5 != 3` actually does, so I think the only one I feel confident enough to name would be `D_801C02E4` (the hookshot one). I propose naming it `sPlayerRightHandFirstPersonHookshotDLs`

Reply:
I did rather assume that based on them being used in the `Player_OverrideLimbDrawGameplayFirstPerson` function.